Resource/PYTHON/multilateration/multilateration.py

```diff
@@ -0,0 +1,287 @@
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STARBOARD_HYD_Y
+                - param.HydrophoneDetails.PORT_HYD_Y,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STARBOARD_HYD_Z
+                - param.HydrophoneDetails.PORT_HYD_Z,
+                2,
+            )
+        )
+    )
+    dist_port_stern = float(
+        math.sqrt(
+            math.pow(
+                param.HydrophoneDetails.STERN_HYD_X
+                - param.HydrophoneDetails.PORT_HYD_X,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Y
+                - param.HydrophoneDetails.PORT_HYD_Y,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Z
+                - param.HydrophoneDetails.PORT_HYD_Z,
+                2,
+            )
+        )
+    )
+    dist_starboard_stern = float(
+        math.sqrt(
+            math.pow(
+                param.HydrophoneDetails.STERN_HYD_X
+                - param.HydrophoneDetails.STARBOARD_HYD_X,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Y
+                - param.HydrophoneDetails.STARBOARD_HYD_Y,
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Z
+                - param.HydrophoneDetails.STARBOARD_HYD_Z,
+                2,
+            )
+        )
+    )
+
+    # Finding the maximum distance
+    max_hydrophone_distance = max(
+        dist_port_starboard, max(dist_starboard_stern, dist_port_stern)
+    )
+
+    # Calculating max time allowed over that distance
+    max_time_diff = (1 + param.SystemMargins.MARGIN_TIME_EPSILON) * (
+        max_hydrophone_distance / param.PhysicalConstants.SOUND_SPEED
+    )
+
+    # Returning if both variables have been set correctly
+    return check_initialized_globals()
+
+
+def initialize_A_matrix():  # return type Matrix_2_3_f
+    A = np.zeros((2, 3))
+    return A
+
+
+def initialize_B_vector():
+    B = np.zeros((2, 1))
+    return B
+
+
+# Functions to check if signals/data are valid
+
+
+def check_initialized_globals():  # return type uint8_t
+    return max_hydrophone_distance != -1 and max_time_diff != 1
+
+
+def check_valid_time(time_lhs: np.uint32, time_rhs: np.uint32):  # return type uint8_t
+
+    # Calculating the time-difference and checking if it exceeds the maximum
+    # allowed time for a valid signal
+
+    time_diff = time_lhs - time_rhs  # type int32_t
+    return (abs(time_diff) * param.DSPConstants.SAMPLE_TIME) > max_time_diff
+
+
+def check_valid_time(time_diff: np.uint32):  # return type uint8_t
+
+    # Checking if the time_diff exceeds the maximum allowed time for a valid signal
+
+    return (abs(time_diff) * param.DSPConstants.SPSAMPLE_TIME) > max_time_diff
```

Comment:
What's going on here? Is this intentionally duplicated with different arguments? Function overloading does not exist and only the latter is valid AFAIK. So either rename the function or remove one.

Isn't this naming wrong anyways? Function checks if time is invalid not valid.